hacspec/src/lib.rs

```diff
@@ -855,46 +884,62 @@ fn parse_message_1(
     EDHOCError,
 > {
     let mut error: EDHOCError = EDHOCError::UnknownError;
+    let mut method: U8 = U8(0xff);
     let mut g_x: BytesP256ElemLen = BytesP256ElemLen::new();
     let mut suites_i = BytesSuites::new();
     let mut suites_i_len: usize = 0;
     let mut raw_suites_len: usize = 0;
     let mut c_i = U8(0);
     let mut ead_1 = None::<EADItemHacspec>;
 
-    let method = rcvd_message_1.content[0];
-
-    let res_suites = parse_suites_i(rcvd_message_1);
+    // check surplus array encoding
+    if !is_cbor_array(rcvd_message_1.content[0]) {
```

Comment:
I don't understand the logic here. You are returning `ParsingError` if it is a CBOR array. What if it is a map? Or something else?

Reply:
Good question. Re-reading the traces document, this should validate that the message is a CBOR sequence, so I think I will instead check for the type of each element.